File: SmartBrowser_java_eclipse/src/com/teuskim/sbrowser/Log.java
package com.teuskim.sbrowser;

public class Log {
	
	private static final boolean IS_LOGGING = false;

	public static void d(String tag, String msg){
		if(IS_LOGGING){
			android.util.Log.d(tag, msg);
		}
	}

	public static void d(String tag, String msg, Throwable tr){
		if(IS_LOGGING){
			android.util.Log.d(tag, msg, tr);
		}
	}

	public static void e(String tag, String msg){
			android.util.Log.e(tag, msg);
	}

	public static void e(String tag, String msg, Throwable tr){
			android.util.Log.e(tag, msg, tr);
	}

	public static void i(String tag, String msg){
		if(IS_LOGGING){
			android.util.Log.i(tag, msg);
		}
	}

	public static void i(String tag, String msg, Throwable tr){
		if(IS_LOGGING){
			android.util.Log.i(tag, msg, tr);
		}
	}

	public static void v(String tag, String msg){
		if(IS_LOGGING){
			android.util.Log.v(tag, msg);
		}
	}

	public static void v(String tag, String msg, Throwable tr){
		if(IS_LOGGING){
			android.util.Log.v(tag, msg, tr);
		}
	}

	public static void w(String tag, String msg){
		if(IS_LOGGING){
			android.util.Log.w(tag, msg);
		}
	}

	public static void w(String tag, String msg, Throwable tr){
		if(IS_LOGGING){
			android.util.Log.w(tag, msg, tr);
		}
	}
	
}

File: SmartBrowser_kotlin/app/src/main/java/com/teuskim/sbrowser/Log.kt
package com.teuskim.sbrowser

object Log {

    private val IS_LOGGING = true

    fun d(tag: String, msg: String) {
        if (IS_LOGGING) {
            android.util.Log.d(tag, msg)
        }
    }

    fun d(tag: String, msg: String, tr: Throwable) {
        if (IS_LOGGING) {
            android.util.Log.d(tag, msg, tr)
        }
    }

    fun e(tag: String, msg: String) {
        android.util.Log.e(tag, msg)
    }

    fun e(tag: String, msg: String, tr: Throwable) {
        android.util.Log.e(tag, msg, tr)
    }

    fun i(tag: String, msg: String) {
        if (IS_LOGGING) {
            android.util.Log.i(tag, msg)
        }
    }

    fun i(tag: String, msg: String, tr: Throwable) {
        if (IS_LOGGING) {
            android.util.Log.i(tag, msg, tr)
        }
    }

    fun v(tag: String, msg: String) {
        if (IS_LOGGING) {
            android.util.Log.v(tag, msg)
        }
    }

    fun v(tag: String, msg: String, tr: Throwable) {
        if (IS_LOGGING) {
            android.util.Log.v(tag, msg, tr)
        }
    }

    fun w(tag: String, msg: String) {
        if (IS_LOGGING) {
            android.util.Log.w(tag, msg)
        }
    }

    fun w(tag: String, msg: String, tr: Throwable) {
        if (IS_LOGGING) {
            android.util.Log.w(tag, msg, tr)
        }
    }

}
